salsa_dancing_molecules/worker_process/simulation_starter.py

```diff
@@ -1,17 +1,47 @@
 """Module for determining which simulation program to be used."""
+from ..simulations.simulation import run
+from asap3 import AsapError
 
 
-def start_simulation(simulation_info, atoms_object):
+def start_simulation(sim_info, atoms):
     """Take data on the simulation and send it to the correct module.
 
     Args:
         simulation_info: Dictionary with information on the simulation.
         atoms_object: Material to be used for the simulation.
     """
-    if (simulation_info["ensemble"] == "nve" and
-            simulation_info["potential"] == "lennard_jones"):
-        from ..simulations import nve
-        nve.run(atoms_object,
-                simulation_info["steps"],
-                simulation_info["output_path"],
-                simulation_info["use_asap"])
+    repeat = int(sim_info["repeat"])
+    symbols = atoms.symbols
+    if repeat > 0:
+        atoms = atoms.repeat(repeat)
+
+    while True:
+        try:
+            print(f'Simulating for {symbols}')
+            run(sim_info, atoms)
+        except ValueError as e:
+            print(f'Size error for {symbols}: {e}')
+            # failsafe against automatic increase of cells becomes too
+            # large
+            if repeat > 10:
+                print('Automatic increase of amount of cells due to too'
+                      'large amount of cells, changing potential to ASAP '
+                      'Lennard-Jones')
+                print('Use --repeat X to manually change to larger amount'
+                      'of cells')
+                sim_info["potential"] = "lennard-jones"
+                run(sim_info, atoms)
+            else:
+                print('Increasing amount of cells')
```

Comment:
It would be nice if this print included what the new repetition value is.

Reply:
Done